### ISSUE TITLE
[deploy] Do not fail Maven Central Rules sources or javadocs jar missing for an empty jar.

### ISSUE THREAD

**Issue by ge0ffrey** [jreleaser/jreleaser#1289]
JReleaser should do the following:

- inspect the JAR looking for `*.class` entries. If found then `-sources` and `-javadoc` is required.
- If not found then look for other entries (besides `META-INF/MANIFEST.MF` and `META-INF/maven/*`). If found then `-sources` is required.

Long discussion: https://github.com/jreleaser/jreleaser/discussions/1287

**Comment by ge0ffrey** [comment]
**Workaround**:

```
              <maven>
                <nexus2>
                  <maven-central>
                    <applyMavenCentralRules>false</applyMavenCentralRules>
                    ...
```

**Comment by ge0ffrey** [comment]
Please reopen this issue.

This issue isn't fixed entirely. While the checksum doesn't fail, it now fails later with 1.6.0-SNAPSHOT:

```
org.jreleaser.model.JReleaserException: Rules for publishing to Maven Central were not met
	at org.jreleaser.sdk.commons.AbstractMavenDeployer.collectDeployables(AbstractMavenDeployer.java:148)
```

```
[ERROR]     [nexus2] timefold-solver-core-0.8.38-sources.jar is missing
[ERROR]     [nexus2] timefold-solver-core-0.8.38-javadoc.jar is missing
[ERROR]     [nexus2] timefold-solver-spring-boot-starter-0.8.38-sources.jar is missing
[ERROR]     [nexus2] timefold-solver-spring-boot-starter-0.8.38-javadoc.jar is missing
[ERROR]     [nexus2] timefold-solver-ide-config-0.8.38-javadoc.jar is missing
```

Reproducer (after running set version to 0.8.38 instead of SNAPSHOT on it): https://github.com/TimefoldAI/timefold-solver/tree/0.8.x

**Comment by ge0ffrey** [comment]
I don't want to spam this thread, so I opened a separate dicussion thread to figure out where the issue lies: https://github.com/jreleaser/jreleaser/discussions/1355

**Comment by aalmiray** [comment]
🎉 This issue has been resolved in `v1.6.0` ([Release Notes](https://github.com/jreleaser/jreleaser/releases/tag/v1.6.0))
